src/opm/parser/eclipse/EclipseState/Schedule/Schedule.cpp

```diff
@@ -484,8 +484,13 @@ namespace Opm {
                 WellCommon::StatusFromString(record.getItem("STATUS").getTrimmedString(0));
 
             auto wells = getWells(wellNamePattern);
+            if (wells.empty()) {
```

Comment:
For simplicity I would say the handling of empty well sets should be in the `getWells( )` method, otherwise you have to duplicate the `if (wells.empty()) { .... }` code block many places.

Reply:
I agree regarding simplicity, and from going through the call sites that should be acceptable. However, if we want to include keyword.name in the user feedback, the function would need `DeckKeyword` as well.. So the function would be
`Schedule::getWells(const std::string& wellNamePattern, const ParseContext& parseContext, const DeckKeyword& keyword )`

Reply:
> However, if we want to include keyword.name in the user feedback, the function would need DeckKeyword [...]

OK I see that; how about:
```C++
if (wells.size() == 0)
    handleWellError( );
```

Reply:
Added `InvalidWellPattern()` (parsecontext uses handleError, thought it would be less ambiguou) as suggested. Rebased and made sure that all commits passes the tests along the way.